py/torch_tensorrt/dynamo/conversion/impl/select.py

```diff
@@ -61,3 +69,271 @@ def select(
     if len(out.shape) != 1:
         layer = network.add_shuffle(out)
     return layer.get_output(0)
+
+
+def index(
+    network: TRTNetwork,
+    target: Target,
+    source_ir: Optional[SourceIR],
+    name: str,
+    input: TRTTensor,
+    index: Union[TRTTensor, Sequence[TRTTensor]],
+) -> TRTTensor:
+    adv_indx_indices = []
+    tensor_indices = []
+    # _LOGGER.debug(f"The index shape is {index.shape}")
+    # check if the input is dynamic
+    dynamic_shape = has_dynamic_shape(input.shape)
+
+    # here we need to check if all the index are broadcastable
+    # if no, then we need to broadcast
+
+    last_index = None
+    broadcast_shape_len = 0
+    for i, ind in enumerate(index):
+        if ind is not None:
+            _LOGGER.debug(f"Shape of {i} index is {ind.shape}")
+            adv_indx_indices.append(i)
+            # torch.nn.parameter.Parameter=> torch.Tensor
+            ind = get_trt_tensor(network, ind, f"parameter_to_fp32_tensor_{i}")
```

Comment:
For layer names, consider switching the name to be prefaced with `_` so the words are separated - as in `_parameter_to_fp32_tensor_{i}`. Additionally, the name has to be appended to the existing layer's `name`, otherwise the layer names will be duplicate across multiple `index` layers in the same model:
```python
name + f"_parameter_to_fp32_tensor_{i}"
```